libsql/src/replication/remote_client.rs

```diff
@@ -7,33 +7,38 @@ use libsql_replication::frame::{Frame, FrameHeader, FrameNo};
 use libsql_replication::meta::WalIndexMeta;
 use libsql_replication::replicator::{map_frame_err, Error, ReplicatorClient};
 use libsql_replication::rpc::replication::{
-    verify_session_token, HelloRequest, LogOffset, SESSION_TOKEN_KEY,
+    verify_session_token, Frames, HelloRequest, LogOffset, SESSION_TOKEN_KEY,
 };
 use tokio_stream::Stream;
 use tonic::metadata::AsciiMetadataValue;
+use tonic::{Response, Status};
 use zerocopy::FromBytes;
 
 /// A remote replicator client, that pulls frames over RPC
 pub struct RemoteClient {
     remote: super::client::Client,
+    secondary_remote: super::client::Client,
```

Comment:
Why does this need a secondary client, why can't we use the other client?

Reply:
because both `self.remote.replication.hello` and `self.remote.replication.batch_log_entries` borrow mut self and rust screems at me when I try to do:
```
let (hello, frames) = tokio::join!(
                self.remote.replication.hello(hello_req),
                self.remote.replication.batch_log_entries(log_offset_req)
            );
```